lib/src/signed_video_h26x_auth.c

```diff
@@ -964,3 +967,42 @@ signed_video_add_nalu_and_authenticate(signed_video_t *self,
 
   return svi_rc_to_signed_video_rc(status);
 }
+
+
+SignedVideoReturnCode
+signed_video_set_public_key(signed_video_t *self, sign_algo_t algo, const char *public_key, size_t public_key_size)
+{
+  if (!self || !public_key || public_key_size == 0) return SV_INVALID_PARAMETER;
+  if (self->signature_info->public_key) return SV_NOT_SUPPORTED;
+
+  // Return SV_NOT_SUPPORTED if not called from start of stream
+  if (self->nalus_in_sei == true) return SV_NOT_SUPPORTED;
+
+  uint8_t *new_public_key = NULL;
+  svi_rc status = SVI_UNKNOWN;
+  SVI_TRY()
+    SVI_THROW_IF_WITH_MSG(
+        algo < 0 || algo >= SIGN_ALGO_NUM, SVI_NOT_SUPPORTED, "Algo is not supported");
+    // Make sure we have allocated enough memory
+    if (self->signature_info->public_key_size != public_key_size) {
```

Comment:
This check, and use of realloc, is not necessary. On line 976 we check for existence of stored public key, hence it will always be a NULL pointer with size 0 at this point.

```
self->signature_info->public_key = malloc(public_key_size);
SVI_THROW_IF(!self->signature_info->public_key, SVI_MEMORY);
```
for this if block should be enough, right?